Core/Business/Addon/Theme/Theme.php

```diff
@@ -75,4 +87,66 @@ public function onReset()
     {
         return true;
     }
+
+    public function setPropertiesFromJson()
+    {
+        $properties = $this->getJsonConfig('theme.json');
+
+        if (is_null($properties)) {
+            throw new \Exception("Missing theme.json for theme $this->directory", 1);
+        }
+
+        $properties->settings = $this->getJsonConfig('settings.json');
+
+        foreach ($properties as $prop => $value) {
+            $this->{$prop} = $value;
+        }
+
+        return $this;
+    }
+
+    public function initSettings()
+    {
+        $this->settings = new \stdClass();
+        $this->setAvailableLayouts();
+
+        return $this->saveSettings();
+    }
+
+    public function setAvailableLayouts()
+    {
+        $layouts = glob($this->directory.'templates/layouts/*.tpl');
```

Comment:
Automatically finding the available layouts is a good idea, but it would be nice to also provide a way for the designers to *describe* the layouts. Perhaps an (optional) key in the theme settings file could provide a human readable name and description for layout files.